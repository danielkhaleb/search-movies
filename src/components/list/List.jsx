import React, { Component } from 'react';

class List extends Component {

    listDetails(source) {
        return (
            <div className="column">
                <div className="card card-movie">
                    <div className="card-image">
                        <figure className="image is-4by3">
                            {
                                source.Poster === undefined || source.Poster === "N/A" ?
                                    // eslint-disable-next-line
                                    <img src={require('../../utils/imgs/image-not-found.gif')} alt="Image not found" />
                                    :
                                    // eslint-disable-next-line 
                                    <img src={source.Poster} alt="Image of movie" />
                            }
                        </figure>
                    </div>
                    <div className="card-content">
                        <div className="media-content">
                            <p className="title is-6">Title: {source.Title}</p>

                        </div>
                        <div class="content">
                            <br />
                            <p className="subtitle is-7">Type: {source.Type}</p>
                            <p className="subtitle is-7">Year: {source.Year}</p>
                            <p className="subtitle is-7">Imdb Id: {source.imdbID}</p>
                        </div>
                    </div>
                </div>
            </div>

        );
    }

    render() {
        const topFive = this.props.data.map((source, index) => {
            if (index < 5) {
                return (this.listDetails(source));
            }
            return null;

        });

        const lastFive = this.props.data.map((source, index) => {
            if (index > 4) {
                return (this.listDetails(source));
            }
            return null;

        });

        return (
            <div>
                <div className="columns">
                    {topFive}
                </div>
                <div className="columns">
                    {lastFive}
                </div>
            </div>
        );
    }
}
export default List;
