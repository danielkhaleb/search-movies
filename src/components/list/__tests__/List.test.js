import React from 'react';
import List from '../List';
import moviesJson from "../../../../mock/movies.json"
import renderer from 'react-test-renderer';

describe('List component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(
            <List
                data={moviesJson}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});