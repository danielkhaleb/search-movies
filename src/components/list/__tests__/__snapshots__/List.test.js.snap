// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders correctly 1`] = `
<div>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <div
        className="card card-movie"
      >
        <div
          className="card-image"
        >
          <figure
            className="image is-4by3"
          >
            <img
              alt="Image not found"
              src="image-not-found.gif"
            />
          </figure>
        </div>
        <div
          className="card-content"
        >
          <div
            className="media-content"
          >
            <p
              className="title is-6"
            >
              Title: 
              Test
            </p>
          </div>
          <div
            class="content"
          >
            <br />
            <p
              className="subtitle is-7"
            >
              Type: 
              Movie
            </p>
            <p
              className="subtitle is-7"
            >
              Year: 
              2019
            </p>
            <p
              className="subtitle is-7"
            >
              Imdb Id: 
              123
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="columns"
  />
</div>
`;
